### PR TITLE
Minor fixes for IMX8MM EVK + 2021.04+imx 5.10.35 2.0.0

### PR DESCRIPTION
**Pull request by ricardosalveti**
Minor fixes for getting a working build for the IMX8MM EVK target.

Removing bootm len, updating SPL DM_I2C and bumping SYS_SPL_MALLOC_SIZE might also be needed for other targets, didn't yet review them (first trying to get a working build for imx8mm to unblock the systemready stuff).

### CODE REVIEW

**Comment by ricardosalveti** [comment]
I also noticed that patch 6b0c6d738367644a8c13aae303caf1d48cd4a3c0 causes a build failure:

```
drivers/crypto/fsl/fsl_rsa.c: In function 'fsl_mod_exp':
drivers/crypto/fsl/fsl_rsa.c:26:29: warning: dereferencing 'void *' pointer
   26 |  fdt64_t exp = fdt64_to_cpu(*prop->public_exponent);
      |                             ^~~~~~~~~~~~~~~~~~~~~~
include/linux/byteorder/swab.h:89:31: note: in definition of macro '__swab64'
   89 | (__builtin_constant_p((__u64)(x)) ? \
      |                               ^
include/linux/byteorder/generic.h:93:21: note: in expansion of macro '__be64_to_cpu'
   93 | #define be64_to_cpu __be64_to_cpu
      |                     ^~~~~~~~~~~~~
drivers/crypto/fsl/fsl_rsa.c:26:16: note: in expansion of macro 'fdt64_to_cpu'
   26 |  fdt64_t exp = fdt64_to_cpu(*prop->public_exponent);
      |                ^~~~~~~~~~~~
include/linux/byteorder/little_endian.h:39:50: error: invalid use of void expression
   39 | #define __be64_to_cpu(x) __swab64((__force __u64)(__be64)(x))
```

**Comment by ricardosalveti** [comment]
Another problem I noticed is that SPL sec_init is failing during SPL init, which depends on FSL_CAAM (and which we don't really need to enable, as we want OP-TEE to use CAAM first to avoid issues with job ring and such).

Looking at why FSL_CAAM is getting enabled, I noticed it is basically being forced via arch/arm/mach-imx/imx8m/Kconfig, and not checking for the HAS_CAAM boolean flag we used before. This is also something we might need to come up with a generic fix for, probably making it flexible enough for being enabled via config instead of being forced via Kconfig.

File: board/freescale/imx8mm_evk/spl.c [hunk 1]
```diff
@@ -326,24 +326,3 @@ void board_init_f(ulong dummy)
 
 	board_init_r(NULL, 0);
 }
-
```
Comment (MrCry0) [inline board/freescale/imx8mm_evk/spl.c:329]: Wow! I've never seen this part of that commit before. Pretty strange :(
Thanks for rechecking!

**Review by MrCry0 — APPROVED** [review]
LGTM

**Comment by MrCry0** [comment]
> I also noticed that patch [6b0c6d7](https://github.com/foundriesio/u-boot/commit/6b0c6d738367644a8c13aae303caf1d48cd4a3c0) causes a build failure:
> 
> ```
> drivers/crypto/fsl/fsl_rsa.c: In function 'fsl_mod_exp':
> drivers/crypto/fsl/fsl_rsa.c:26:29: warning: dereferencing 'void *' pointer
>    26 |  fdt64_t exp = fdt64_to_cpu(*prop->public_exponent);
>       |                             ^~~~~~~~~~~~~~~~~~~~~~
> include/linux/byteorder/swab.h:89:31: note: in definition of macro '__swab64'
>    89 | (__builtin_constant_p((__u64)(x)) ? \
>       |                               ^
> include/linux/byteorder/generic.h:93:21: note: in expansion of macro '__be64_to_cpu'
>    93 | #define be64_to_cpu __be64_to_cpu
>       |                     ^~~~~~~~~~~~~
> drivers/crypto/fsl/fsl_rsa.c:26:16: note: in expansion of macro 'fdt64_to_cpu'
>    26 |  fdt64_t exp = fdt64_to_cpu(*prop->public_exponent);
>       |                ^~~~~~~~~~~~
> include/linux/byteorder/little_endian.h:39:50: error: invalid use of void expression
>    39 | #define __be64_to_cpu(x) __swab64((__force __u64)(__be64)(x))
> ```

Arrgh, sorry, this commit shouldn't have been included in the PR :(
It was only as an example of reworking Jorge's one.

Fixed.

**Comment by MrCry0** [comment]
> Another problem I noticed is that SPL sec_init is failing during SPL init, which depends on FSL_CAAM (and which we don't really need to enable, as we want OP-TEE to use CAAM first to avoid issues with job ring and such).
> 
> Looking at why FSL_CAAM is getting enabled, I noticed it is basically being forced via arch/arm/mach-imx/imx8m/Kconfig, and not checking for the HAS_CAAM boolean flag we used before. This is also something we might need to come up with a generic fix for, probably making it flexible enough for being enabled via config instead of being forced via Kconfig.

Ok, I'm handling it today.

**Review by MrCry0 — APPROVED** [review]
Ricardo, the fixes are good, I used your work and placed my own [fixes](https://github.com/foundriesio/u-boot/pull/97) on top of yours, improving only one of your commit messages.

Which variant to merge is up to you. If you merge your ones, I'll rebase mine again.

**Comment by ricardosalveti** [comment]
Closing in favor of #97 .

**Pull request closed without merging**